packages/accordion/src/useAccordion.ts

```diff
@@ -5,185 +5,150 @@
-  expandedSections: number[];
-  disabledSections: number[];
-}
+import {
+  IUseAccordionProps,
+  IUseAccordionReturnValue,
+  IHeaderProps,
+  ITriggerProps,
+  IPanelProps
+} from './types';
 
 export function useAccordion({
   idPrefix,
+  sections,
   expandedSections,
+  defaultExpandedSections,
   onChange,
   expandable = true,
-  collapsible = true,
-  defaultExpandedSections
+  collapsible = true
 }: IUseAccordionProps = {}): IUseAccordionReturnValue {
-  const isControlled = expandedSections !== null && expandedSections !== undefined;
-  const seed = useUIDSeed();
-  const prefix = useMemo<string>(
-    () => idPrefix || seed(`accordion_${PACKAGE_VERSION}`),
-    [idPrefix, seed]
-  );
+  const prefix = useId(idPrefix);
   const TRIGGER_ID = `${prefix}--trigger`;
   const PANEL_ID = `${prefix}--panel`;
+
+  const isControlled = expandedSections !== null && expandedSections !== undefined;
   const [expandedState, setExpandedState] = useState<number[]>(defaultExpandedSections || [0]);
+  const [disabledState, setDisabledState] = useState(collapsible ? [] : expandedState);
+  const internalExpandedState = getControlledValue(expandedSections, expandedState)!;
 
-  const controlledExpandedState = getControlledValue(expandedSections, expandedState)!;
+  const toggle = useCallback(
+    (index: number) => {
+      const expanded: number[] = [];
+      const disabled: number[] = [];
 
-  const [disabledState, setDisabledState] = useState(collapsible ? [] : expandedState);
+      sections!.forEach(sectionIndex => {
+        let isExpanded = false;
 
-  const sectionIndices: number[] = [];
-  const toggle = (index: number) => {
-    const expanded: number[] = [];
-    const disabled: number[] = [];
+        if (sectionIndex === index) {
+          isExpanded = collapsible ? expandedState.indexOf(sectionIndex) === -1 : true;
+        } else if (expandable) {
+          isExpanded = expandedState.indexOf(sectionIndex) !== -1;
+        }
+
+        if (isExpanded) {
+          expanded.push(sectionIndex);
 
-    sectionIndices.forEach(sectionIndex => {
-      let isExpanded = false;
+          if (!collapsible) {
+            disabled.push(sectionIndex);
+          }
+        }
+      });
 
-      if (sectionIndex === index) {
-        isExpanded = collapsible ? expandedState.indexOf(sectionIndex) === -1 : true;
-      } else if (expandable) {
-        isExpanded = expandedState.indexOf(sectionIndex) !== -1;
+      if (onChange) {
+        onChange(index);
       }
 
-      if (isExpanded) {
-        expanded.push(sectionIndex);
+      if (isControlled === false) {
+        setExpandedState(expanded);
+      }
 
-        if (!collapsible) {
-          disabled.push(sectionIndex);
-        }
+      setDisabledState(disabled);
+    },
+    [sections, expandedState, collapsible, expandable, isControlled, onChange]
+  );
+
+  const getHeaderProps = useCallback(
+    ({ role = 'heading', ariaLevel, ...props }: IHeaderProps = {}) => {
```

Comment:
Kind of not the fault of this PR, but I'm curious anyway... any reason we don't use the `aria-level` attribute directly? If we're breaking the API anyway, this feels like a reasonable additional change.

```suggestion
    ({ role = 'heading', 'aria-level': ariaLevel, ...props }: IHeaderProps = {}) => {
```

Only bringing it up because I see we use `aria-*` attributes this way in other containers/components.

Reply:
This makes sense to me. I've updated.